eviatar/enable_windows_features.ps1

```diff
@@ -0,0 +1,33 @@
+﻿function Install-WindowsFeatures {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string[]]$Features
+    )
+    
+    foreach ($Feature in $Features) {
+        Write-Host "Installing feature: $Feature"
+        Install-WindowsFeature -Name $Feature -IncludeAllSubFeature -IncludeManagementTools -Verbose
+    }
+}
+
+function Rename-Server {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$NewName
+    )
+
+    Write-Host "Renaming server to: $NewName"
+    Rename-Computer -NewName $NewName -Force

```

Comment:
Why wrap a function in another function just to add a log line?
What happens if the Rename-Computer fails?

Reply:
You right, Instead, I can directly include the logging in the main code or use a Try-Catch block to handle errors during renaming. If Rename-Computer fails, handling it within a Catch block allows for logging and corrective actions.